src/core/p5.Renderer.js

```diff
@@ -13,347 +13,395 @@ import * as constants from '../core/constants';
  * Renderer2D and Renderer3D classes, respectively.
  *
  * @class p5.Renderer
- * @constructor
- * @extends p5.Element
- * @param {String} elt DOM node that is wrapped
- * @param {p5} [pInst] pointer to p5 instance
- * @param {Boolean} [isMainCanvas] whether we're using it as main canvas
  */
-p5.Renderer = function (elt, pInst, isMainCanvas) {
-  p5.Element.call(this, elt, pInst);
-  this.canvas = elt;
-  this._pixelsState = pInst;
-  if (isMainCanvas) {
-    this._isMainCanvas = true;
-    // for pixel method sharing with pimage
-    this._pInst._setProperty('_curElement', this);
-    this._pInst._setProperty('canvas', this.canvas);
-    this._pInst._setProperty('width', this.width);
-    this._pInst._setProperty('height', this.height);
-  } else {
-    // hide if offscreen buffer by default
-    this.canvas.style.display = 'none';
-    this._styles = []; // non-main elt styles stored in p5.Renderer
-  }
+class Renderer extends p5.Element {
+  /**
+    * @class p5.Renderer
+    * @constructor
+    * @extends p5.Element
+    * @param {String} elt DOM node that is wrapped
+    * @param {p5} [pInst] pointer to p5 instance
+    * @param {Boolean} [isMainCanvas] whether we're using it as main canvas
+    */
```

Comment:
Just to note here as with p5.Element, these documentation should probably be left in the original location or it might not render correctly.